x/group/keeper/invariants.go

```diff
@@ -71,35 +71,38 @@ func GroupTotalWeightInvariantHelper(ctx sdk.Context, storeService storetypes.KV
 			return msg, broken
 		}
 
-		memIt, err := groupMemberByGroupIndex.Get(kvStore, groupInfo.Id)
-		if err != nil {
-			msg += fmt.Sprintf("error while returning group member iterator for group with ID %d\n%v\n", groupInfo.Id, err)
-			return msg, broken
-		}
-		defer memIt.Close()
-
-		for {
-			var groupMember group.GroupMember
-			_, err = memIt.LoadNext(&groupMember)
-			if errors.ErrORMIteratorDone.Is(err) {
-				break
-			}
-			if err != nil {
-				msg += fmt.Sprintf("LoadNext failure on member table iterator\n%v\n", err)
-				return msg, broken
-			}
-
-			curMemWeight, err := groupmath.NewPositiveDecFromString(groupMember.GetMember().GetWeight())
+		err = func() error {
+			memIt, err := groupMemberByGroupIndex.Get(kvStore, groupInfo.Id)
 			if err != nil {
-				msg += fmt.Sprintf("error while parsing non-nengative decimal for group member %s\n%v\n", groupMember.Member.Address, err)
-				return msg, broken
+				return fmt.Errorf("error while returning group member iterator for group with ID %d\n%v", groupInfo.Id, err)
 			}
-
-			membersWeight, err = groupmath.Add(membersWeight, curMemWeight)
-			if err != nil {
-				msg += fmt.Sprintf("decimal addition error while adding group member voting weight to total voting weight\n%v\n", err)
-				return msg, broken
+			defer memIt.Close()
+
+			for {
+				var groupMember group.GroupMember
+				_, err = memIt.LoadNext(&groupMember)
+				if errors.ErrORMIteratorDone.Is(err) {
+					break
```

Comment:
personal preference: break is good and does the job but you can also do `return nil` here already to exit early. There is nothing going on after the loop and a return would make it slightly more readable.